test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,169 @@
+  let copy_dir src dst =
+    assert(not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" ["-R";src;dst] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Failed to set up the test env; command `%s' exited \
+                    with non-zero code %d\n" cmd n
+
+  (** More specific utils from here *)
+
+  let v1_store_archive_dir = "test/irmin-pack/data/version_1"
+
+  (** Check we are in the location we expect, the project root. See
+     test_existing_stores.goto_project_root for alternative. *)
+  let assert_project_root () =
+    match Sys.file_exists v1_store_archive_dir with
+    | true -> ()
+    | false ->
+      Fmt.failwith "Couldn't find dir %S from cwd %S\n%!"
+        v1_store_archive_dir
+        Filename.current_dir_name
+
+  let _ = assert_project_root ()
+
+  module Unix_ = Irmin_pack.IO.Unix
+
+  (** Get the version of the underlying file; file is assumed to
+     exist; file is assumed to be an Irmin_pack.IO.Unix file *)
+  let io_get_version ~fn : [`V1 | `V2] =
+    assert(Sys.file_exists fn);
+    Unix_.v ~version:None ~fresh:false ~readonly:true fn |> fun t ->
+    let r = Unix_.version t in
+    Unix_.close t;
+    r
+
+end
+open Util
+
+
+(** This sets up infrastructure to open the existing "version_1" store *)
+module With_existing_store() = struct
+
+  let tmp_dir = tmp_dir ()
+
+  let _ = [%log.info "Using temporary directory %s\n%!" tmp_dir]
+
+  (* Make a copy of the v1_store_archive_dir in tmp_dir *)
+  let _ =
+    assert_project_root();
+    rm_dir tmp_dir;
+    copy_dir v1_store_archive_dir tmp_dir;
+    ()
+
+  (** Set up modules to allow access to "version_1" store *)
+  module Store_ = struct
+    (* In test_existing_stores.ml, we have Test_reconstruct, which
+       uses S.traverse_pack_file; so S is likely the correct config
+       for the pack file. S = V2(), V2() = V2_maker.Make(Schema);
+       V2_maker = Irmin_pack.Maker(Conf); Conf is from Tezos_irmin;
+       Schema is from common *)
+    module Conf = Irmin_tezos.Conf
+
+    module Schema = Common.Schema
+    (* same as Irmin_tezos.Schema? no; tezos uses their own hash;
+       common has Hash.SHA1 *)
+
+    (* from test_existing_stores.ml; FIXME why V2? *)
+    module V2_maker = Irmin_pack.Maker (Conf)
+    module V2 = V2_maker.Make (Schema)
+  end
+
+  (* [S] and [config] are the main things we use from this module *)
+  module S = Store_.V2
+
+  (* Code copied and modified from test_existing_stores.ml; FIXME is
+     this the config of index, or pack? or both? *)
+  let config ~readonly : Irmin.config =
+    Irmin_pack.config ~readonly ~index_log_size:1000 ~fresh:false tmp_dir
+end
+
+let test_RO_no_version_bump () : unit Lwt.t =
+  [%log.info "Executing test_RO_no_version_bump in dir\n%!"];
+  let open With_existing_store() in
+  assert(io_get_version ~fn:(tmp_dir ^"/store.pack") = `V1);
```

Comment:
We generally favour `Alcotest` assertions rather than plain `assert` since the failures print both expected and actual outputs (and test failures can be difficult to reproduce).

We could do:

```suggestion
  let fn = tmp_dir ^ "/store.pack" in
  check_version __POS__ ~expected:`V2 (io_get_version ~fn);
```

with `check_version` is defined as:

```ocaml
let check_version pos ~expected actual =
  Alcotest.check_repr ~pos Irmin_pack.Version.t "" expected actual
```

(Will need to do `module Version = Version` in `irmin_pack_intf.ml` to expose the typerep, but worth it IMO.)